ext/spl/spl_iterators.c

```diff
@@ -2181,10 +2165,12 @@ static HashTable *spl_dual_it_get_gc(zend_object *obj, zval **table, int *n)
 			break;
 		case DIT_CallbackFilterIterator:
 		case DIT_RecursiveCallbackFilterIterator:
-			if (object->u.cbfilter) {
-				zend_get_gc_buffer_add_zval(gc_buffer, &object->u.cbfilter->fci.function_name);
-				if (object->u.cbfilter->fci.object) {
-					zend_get_gc_buffer_add_obj(gc_buffer, object->u.cbfilter->fci.object);
+			if (ZEND_FCC_INITIALIZED(object->u.callback_filter)) {
+				if (object->u.callback_filter.object) {
+					zend_get_gc_buffer_add_obj(gc_buffer, object->u.callback_filter.object);
+				}
+				if (object->u.callback_filter.closure) {
+					zend_get_gc_buffer_add_obj(gc_buffer, object->u.callback_filter.closure);
 				}
```

Comment:
There is possibly a use-case for a `zend_fcc_gc` function

Reply:
I tried to add it to ``zend_gc.h`` but got some unresolved type issue :/ and I don't understand how it has no issue finding ``HashTable`` and ``zend_object``

Reply:
It makes sense to define the function outside of zend_gc.h :+1: This is what we do for the `get_gc` object handler for instance.